coral-schema/src/test/java/com/linkedin/coral/schema/avro/MergeHiveSchemaWithAvroTests.java

```diff
@@ -0,0 +1,320 @@
+
+    assertSchema(avro, merge(hive, avro));
+  }
+
+  @Test
+  public void shouldRetainDefaultValuesFromAvro() {
+    String hive = "struct<fa:int,fb:struct<ga:int>>";
+    Schema avro = struct("r1", field("fA", Schema.create(Schema.Type.INT), null, 1, null),
+        field("fB", struct("r2", field("gA", Schema.create(Schema.Type.INT), null, 2, null)), null,
+            new TextNode("{\"gA\": 3}"), null));
+
+    assertSchema(avro, merge(hive, avro));
+  }
+
+  @Test
+  public void shouldRetainFieldPropsFromAvro() {
+    String hive = "struct<fa:int,fb:struct<ga:int>>";
+    Schema avro = struct("r1", required("fA", Schema.Type.INT, null, null, ImmutableMap.of("pfA", "vfA")),
+        required("fB", struct("r2", required("gA", Schema.Type.INT, null, null, ImmutableMap.of("pfB", "vfB"))), null,
+            null, ImmutableMap.of("pgA", "vgA")));
+
+    assertSchema(avro, merge(hive, avro));
+  }
+
+  @Test
+  public void shouldHandleLists() {
+    String hive = "struct<fa:array<int>,fb:array<int>,fc:array<struct<ga:int>>,fd:array<int>>";
+    Schema avro = struct("r1", required("fA", array(Schema.Type.INT)), optional("fB", array(Schema.Type.INT)),
+        required("fC", array(struct("r2", required("gA", Schema.Type.INT)))));
+
+    Schema expected = struct("r1", required("fA", array(Schema.Type.INT)), optional("fB", array(Schema.Type.INT)),
+        required("fC", array(struct("r2", required("gA", Schema.Type.INT)))),
+        // Array element type is also nullable because it is generated from Hive
+        optional("fd", array(nullable(Schema.Type.INT))));
+
+    assertSchema(expected, merge(hive, avro));
+  }
+
+  @Test
+  public void shouldHandleMaps() {
+    String hive = "struct<fa:map<string,int>,fb:map<string,int>,fc:map<string,struct<ga:int>>,fd:map<string,int>>";
+    Schema avro = struct("r1", required("fA", map(Schema.Type.INT)), optional("fB", map(Schema.Type.INT)),
+        required("fC", map(struct("r2", required("gA", Schema.Type.INT)))));
+
+    Schema expected = struct("r1", required("fA", map(Schema.Type.INT)), optional("fB", map(Schema.Type.INT)),
+        required("fC", map(struct("r2", required("gA", Schema.Type.INT)))),
+        // Map value type is also nullable because it is generated from Hive
+        optional("fd", map(nullable(Schema.Type.INT))));
+
+    assertSchema(expected, merge(hive, avro));
+  }
+
+  @Test
+  public void shouldSanitizeIncompatibleFieldNames() {
+    StructTypeInfo typeInfo = (StructTypeInfo) TypeInfoFactory.getStructTypeInfo(Lists.newArrayList("a.b.c", "$#@%!"),
+        Lists.newArrayList(TypeInfoFactory.intTypeInfo, TypeInfoFactory.intTypeInfo));
+    Schema avro = struct("r1");
+
+    Schema expected =
+        struct("r1", optional("a_x2Eb_x2Ec", Schema.Type.INT), optional("_x24_x23_x40_x25_x21", Schema.Type.INT));
+    assertSchema(expected, merge(typeInfo, avro));
+  }
+
+  @Test
+  public void shouldReorderOptionalSchemaToMatchDefaultValue() {
```

Comment:
Can we add another test case that we need to reorder the option? 
like
```
{
      type:[null, int],
      default: 1
}
```

Reply:
Added test case for reordering.